### ISSUE TITLE
Apollo gateway should gracefully handle exceptions and not fail the entire operation when of the backend service is down/or throws an error

### ISSUE THREAD

**Issue by Bhoomikapanwar** [apollographql/apollo-server#4555]
Apollo gateway fails the entire operation/request in case any of the backend service is down/or throws an error. The query planner says the queries are executed in parallel and if any those fails, it fails the entire request. This is not something very ideal, there should be graceful exception handling in place so that instead of failing the entire operation, it should return error just for that particular query relating to down service and data for others.

**Comment by abernix** [comment]
Duplicate of https://github.com/apollographql/apollo-server/issues/4424.  The intention was for this not to be the case, so if you can dig into why that expectation isn't being met, your research would be greatly appreciated.  However, I'll close this in favor of that existing issue.

**Comment by Bhoomikapanwar** [comment]
Sure, I debugged the issue and will add my findings to the originally created issue  #4424 
Thank you!!
